### PR TITLE
feat(reddit): add `sanitize-sharing-links` patch

### PR DESCRIPTION
**Pull request by kazimmt**
Credits: @inotia00
https://github.com/inotia00/revanced-patches/commit/42b25fedf9895eb15475b1f0345ebe42c4a56b70

## without parameters removed:
https://www.reddit.com/r/revancedapp/comments/11r4xyu/resolution_on_recent_playback_issues_on_youtube/?utm_source=share&utm_medium=android_app&utm_name=androidcss&utm_term=1&utm_content=share_button

## with parameters removed:
https://www.reddit.com/r/revancedapp/comments/11r4xyu/resolution_on_recent_playback_issues_on_youtube/

## Integration Changes
https://github.com/revanced/revanced-integrations/pull/407

### CODE REVIEW

**Comment by oSumAtrIX** [comment]
The patch features a huge block of instructions that should be outsourced to integrations.  Refer to third point: https://github.com/revanced/revanced-patches/blob/docs/docs/4_structure_and_conventions.md#-conventions.

**Comment by kazimmt** [comment]
> I've refactored the code, but the patch features a huge block of instructions that should be outsourced to integrations. 
I'm unable to move it to the integrations :(

**Comment by oSumAtrIX** [comment]
Please explain your problem.

File: src/main/kotlin/app/revanced/patches/reddit/misc/tracking/url/patch/SanitizeUrlQueryPatch.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+                val urlRegister = instruction(endIndex).registerA
+
+                val insertIndex = endIndex + 1 // Right after URL is moved into a register.
+                addInstructions(
+                    insertIndex,
+                    """
+                        invoke-static {v$urlRegister}, Lapp/revanced/reddit/privacy;->removeTrackingParameters(Ljava/lang/String;)Ljava/lang/String;
+  move-result-object v$urlRegister
+                        """
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/reddit/misc/tracking/url/patch/SanitizeUrlQueryPatch.kt:46]:
```suggestion
                    """
                        invoke-static {v$urlRegister}, Lapp/revanced/reddit/patches/SanitizeUrlQueryPatch;->removeTrackingParameters(Ljava/lang/String;)Ljava/lang/String;
  move-result-object v$urlRegister
                   """
```

**Comment by inotia00** [comment]
It looks like there is no `@RequiresIntegrations`, is that okay?

**Comment by oSumAtrIX** [comment]
No, theres couple issues that need to be fixed.

File: src/main/kotlin/app/revanced/patches/reddit/misc/tracking/url/patch/SanitizeUrlQueryPatch.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                   """
+                )
+            }
+        } ?: return ShareLinkFactoryFingerprint.toErrorResult()
+
+        return PatchResultSuccess()
+    }
+
+    private companion object {
+        val Instruction.registerA
```
Comment (johnconner122) [inline src/main/kotlin/app/revanced/patches/reddit/misc/tracking/url/patch/SanitizeUrlQueryPatch.kt:57]: I might be wrong but isn't it dead code?

**Pull request merged**